### PR TITLE
refactor(adt): updates from SDK board feedback

### PR DESCRIPTION
**Pull request by drwill-ms**
- Flatten namespace by moving type defs under *.Models to the root
- Do not return `Response<IReadOnlyList<T>>`. Changed to `Response<T[]>`.

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs [hunk 1]
```diff
@@ -1190,9 +1190,10 @@ public virtual Response DecommissionModel(string modelId, CancellationToken canc
         /// Console.WriteLine($&quot;Created models &apos;{componentModelId}&apos; and &apos;{sampleModelId}&apos;.&quot;);
         /// </code>
         /// </example>
-        public virtual Task<Response<IReadOnlyList<ModelData>>> CreateModelsAsync(IEnumerable<string> models, CancellationToken cancellationToken = default)
+        public virtual async Task<Response<ModelData[]>> CreateModelsAsync(IEnumerable<string> models, CancellationToken cancellationToken = default)
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1193]: `async` keyword here?

Comment (abhipsaMisra) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1193]: ah, got it, never mind

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/autorest.md [hunk 2]
```diff
@@ -6,5 +6,14 @@ Run `generate.ps1` in this directory to generate the code.
 > see https://aka.ms/autorest
 
 ``` yaml
+#when generating from local changes:
 input-file: $(this-folder)/swagger/digitaltwins.json
+
+#when generating from official source
+#require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/c3dd5df2863101b56eef256b810927cdcc4e44d2/specification/digitaltwins/data-plane/readme.md
```
Comment (abhipsaMisra) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/autorest.md:13]: +1

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs (hunk 1 above)
Comment (abhipsaMisra) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1193]: Just to refresh my memory, what was the service reasoning for returning a non-pageable models array for the "create" operation, vs returning a pageable models array for "get" operation?

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs (hunk 1 above)
Comment (drwill-ms) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs:1193]: We never asked. I posed the question to Brian Crawford this morning.

My guess is that in listing models, you'd potentially expect quite a few, making paging necessary. Also they use CosmosDB under the covers, which gives them paging for free.

However, model creation probably won't have that many models in the payload, so paging really shouldn't be necessary. Also, they probably can't rely on CosmosDB's paging to save them here.

**Comment by KrzysztofCwalina** [comment]
API changes look good.

**Pull request merged**